src/rust/engine/src/console_ui.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+use crate::scheduler::RootResult;
+use indexmap::IndexMap;
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+use logging::logger::LOGGER;
+use parking_lot::Mutex;
+use std::sync::{mpsc, Arc, Barrier};
+use std::time::Duration;
+use workunit_store::WorkUnitStore;
+
+enum ConsoleMessage {
+  FinalResult,
+  //Stdout(String),
+  Stderr(String),
+  DisableUI,
+}
+
```

Comment:
As mentioned above, should probably move this into the `ui` crate to prevent graph cycles.